animation_nodes/algorithms/mesh_generation/find_shortest_path.pyx

```diff
@@ -0,0 +1,112 @@
+from ... data_structures import GPStroke
+from ... data_structures cimport (
+    Mesh,
+    LongList,
+    ColorList,
+    FloatList,
+    PolySpline,
+    DoubleList,
+    BooleanList,
+    Vector3DList,
+)
+
+from libc.math cimport INFINITY
+from . line import getLinesMesh
+from ... math cimport distanceVec3
+
+# Dijkstra's algorithm (https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm) but is implemented
+# such a way to handle multi-sources and mesh with multiple islands.
+def getShortestPath(Mesh mesh, LongList sources, LongList targets, str pathType, str mode):
```

Comment:
Why is `targets` a list?

Reply:
Removed.

Reply:
However, with `targets` a list we can extend the functionality for target-indices, and outputs tree.

Reply:
Okay, but lets finish an initial version first then look at possible ways to extend it.

Reply:
Okay.